src/coreclr/nativeaot/System.Private.CoreLib/src/System/WeakReference.NativeAot.cs

```diff
@@ -35,109 +34,56 @@ private void Create(object? target, bool trackResurrection)
 
         //Determines whether or not this instance of WeakReference still refers to an object
         //that has not been collected.
-        //
         public virtual bool IsAlive
         {
             get
             {
-                IntPtr h = m_handle;
-
-                // In determining whether it is valid to use this object, we need to at least expose this
-                // without throwing an exception.
-                if (default(IntPtr) == h)
-                    return false;
-
-                bool result = (RuntimeImports.RhHandleGet(h) != null || TryGetComTarget() != null);
-
-                // We want to ensure that if the target is live, then we will
-                // return it to the user. We need to keep this WeakReference object
-                // live so m_handle doesn't get set to 0 or reused.
-                // Since m_handle is volatile, the following statement will
-                // guarantee the weakref object is live till the following
-                // statement.
-                return (m_handle == default(IntPtr)) ? false : result;
+                return Target != null;
```

Comment:
This is subtle change. `Target` property is virtual and it call be overridden to do whatever in the inherited type.

Reply:
I did not even think the `Target` could be virtual. `GetGeneration` makes even less sense if `Target` is overridden.

I guess I will revert to the original implementation, except the `KeepAlive(wo);`. We do not need to protect the handle once we fetched the obj. The handle can be gone by the time we return to the caller anyways.

Reply:
The old code also does not do `RuntimeImports.RhHandleGet(h) ?? TryGetComTarget()`. Another difference.

Reply:
Oops, sorry. I confused `IsAlive` and `GC.GetGeneration`. Both have subtle changes if `Target` is overridden. 
I'll revert both to the old implementation.